cylc/flow/task_pool.py

```diff
@@ -1600,62 +1605,184 @@ def spawn_task(
+            flow_descr: description of new flow
 
         """
-        outputs = outputs or [TASK_OUTPUT_SUCCEEDED]
-        if flow_num is None:
-            flow_nums = None
+        flow_nums = self._get_flow_nums(flow, flow_descr)
+        if flow_nums is None:
+            # Illegal flow command opts
+            return
+
+        # Get matching pool tasks and future task definitions.
+        itasks, future_tasks, unmatched = self.filter_task_proxies(
+            items,
+            future=True,
+            warn=False,
+        )
+
+        for itask in itasks:
+            self.merge_flows(itask, flow_nums)
+            if prereqs:
+                self._set_prereqs_itask(
+                    itask, prereqs, flow_nums, flow_wait)
+            else:
+                self._set_outputs_itask(itask, outputs)
+
+        for name, point in future_tasks:
+            tdef = self.config.get_taskdef(name)
+            if prereqs:
+                self._set_prereqs_tdef(
+                    point, tdef, prereqs, flow_nums, flow_wait)
+            else:
+                trans = self._spawn_transient(
+                    point, tdef, flow_nums, flow_wait)
+                if trans is not None:
+                    self._set_outputs_itask(trans, outputs)
+
+        if self.compute_runahead():
+            self.release_runahead_tasks()
+
+    def _set_outputs_itask(
+        self,
+        itask: 'TaskProxy',
+        outputs: Optional[List[str]],
+    ) -> None:
+        """Set requested outputs on a task and spawn associated children."""
+
+        # Default to required outputs.
+        outputs = outputs or itask.tdef.get_required_outputs()
+
+        changed = False
+        for output in outputs:
+            # convert trigger label to output message
+            out = itask.state.outputs.get_msg(output)
+            info = f'set: output {itask.identity}:{output}'
+            if out is None:
+                LOG.warning(f"{info} not found")
+                continue
+            if itask.state.outputs.is_completed(out):
+                LOG.info(f"{info} completed already")
+                continue
+            changed = True
+            self.task_events_mgr.process_message(
+                itask, logging.INFO, out, forced=True)
+            LOG.info(f"{info} completed")
+
+        if changed and itask.transient:
+            # (note tasks states table gets updated from the task pool)
+            self.workflow_db_mgr.put_update_task_state(itask)
+
+        # testing...
+        # self.data_store_mgr.delta_task_outputs(itask)
+        # self.data_store_mgr.delta_task_state(itask)
+        # self.data_store_mgr.update_data_structure()
+        # self.data_store_mgr.prune_data_store()
+        # self.data_store_mgr.update_updates_pending = True
+
+    def _set_prereqs_itask(self, itask, prereqs, flow_nums, flow_wait):
+        """Set prerequisites on a task in the pool."""
+        if prereqs == ["all"]:
```

Comment:
Not sure whether `['all', 'succeeded']` could get this far, if not, can protect against it:

```suggestion
        if 'all' in prereqs:
```

Reply:
It can't - no point in having "all" and specific prerequisites at the same time. But I'll take that just in case we change the front end later to accept both (I guess it's not technically wrong, even if pointless).